inst/odin/carehomes.R

```diff
@@ -356,37 +363,81 @@ n_II_P[, , , ] <- n_I_P_progress[i, j, k, l] -
-  p_I_P_next_vacc_class[i, j, k, l])
-
+  p_I_P_next_vacc_class[i, j, k, l]
+)
 
 #### flow out of R ####
 
-n_R_progress_tmp[, , ] <- rbinom(R[i, j, k], p_RS[i])
+## prob of progress to S or E (for strains 1 and 2)
+##  R1 and R2 can progress to S (w.p. waning_rate[i]) or
+##  R1 can progress to E3 (w.p. lambda[i, 2], dep. on prob. strain 2)
+##  R2 can progress tp E4 (w.p. lambda[i, 1], dep. on prob. strain 1)
+## R3 and R4 can progress to S only (w.p. waning_rate[i])
+p_R_progress[, ] <- if (n_strains == 1) 1 - exp(-waning_rate[i] * dt) else
+  (if (j == 1) 1 - exp(- (waning_rate[i] + lambda[i, 2]) * dt) else
+    (if (j == 2) 1 - exp(- (waning_rate[i] + lambda[i, 1]) * dt) else
+    (1 - exp(-waning_rate[i] * dt))))
+
+## Total number who can possibly progress
+n_R_progress_tmp[, , ] <- rbinom(R[i, j, k], p_R_progress[i, j])
+
 ## cap on people who can move out of R based on numbers in T_sero_neg and
 ## T_PCR_neg
 n_R_progress_capped[, , ] <-
   min(n_R_progress_tmp[i, j, k], T_sero_neg[i, j, k], T_PCR_neg[i, j, k])
+
 ## use cap or not depending on model_pcr_and_serology value
+## n_R_progress is total number who either:
+##  - leave R for S or E and stay in same vacc class
+##  - leave R for S or E and change same vacc class
 n_R_progress[, , ] <- if (model_pcr_and_serology == 1)
   n_R_progress_capped[i, j, k] else n_R_progress_tmp[i, j, k]
-## of those some can also be vaccinated or progress through vaccination classes
-## --> number transitioning from R[j] to S[j+1]
-## (j vaccination class)
-n_RS_next_vacc_class[, , ] <-
-  rbinom(n_R_progress[i, j, k], p_R_next_vacc_class[i, j, k])
-## resulting transitions from R[j] to S[j]
-## (j vaccination class)
-n_RS[, , ] <- n_R_progress[i, j, k] - n_RS_next_vacc_class[i, j, k]
 
-## vaccine progression
-n_R_next_vacc_class_tmp[, , ] <- rbinom(
-  R[i, j, k] - n_R_progress[i, j, k], p_R_next_vacc_class[i, j, k])
-n_R_next_vacc_class_capped[, , ] <-
-  min(n_R_next_vacc_class_tmp[i, j, k],
+## Number going from R to S
+## In one-strain model, all progress to S only
+## In multi-strain model, R3 and R4 progress to S only
+## In multi-strain model, number of R1 and R2 to S is binomial w.p. waning over
+##  waning plus lambda
+## TODO (RS): waning_rate should eventually be variant varying
+n_RS[, , ] <- if (n_strains == 1) n_R_progress[i, j, k] else
+  (if (j == 1) rbinom(n_R_progress[i, j, k],
+                      waning_rate[i] / (waning_rate[i] + lambda[i, 2])) else
+  (if (j == 2) rbinom(n_R_progress[i, j, k],
+                      waning_rate[i] / (waning_rate[i] + lambda[i, 1])) else
+  (n_R_progress[i, j, k])))
+
+## Numbers going from R to E
+##   R -> E1 and E2 are always zero
+##   nR1E3 are Number leaving R1 and not going to S;
+##     nR2E3 eq nR3E3 eq n_R4E3 eq 0;
+##   nR2E4 is Number leaving R2 and not going to S;
+##     nR2E4 gt 0; nR1E4 eq n_R3E4 eq n_R4E4 eq 0;
+## Total possible to E3 and E4 are from R1 and R2 respectively
+n_RE[, , ] <- if (j < 3) 0 else
```

Comment:
would be good to clarify how n_RE is indexed, it looks like n_RE[i, j, k] is the number of individuals moving from R[i,j-2,k] to E[I,j,k] so that it is indexed by the recipient E compartment (although I may be misunderstanding this).